lib/transmuxer.js

```diff
@@ -1029,61 +1142,27 @@ Transmuxer = function() {
   this.metadataStream = new muxjs.mp2t.MetadataStream();
 
   // assemble MPEG2-TS packets into elementary streams
-  packetStream.pipe(parseStream);
-  parseStream.pipe(elementaryStream);
+  packetStream
+    .pipe(parseStream)
+    .pipe(elementaryStream);
+
   // remux the streams
   elementaryStream.pipe(aacStream);
   elementaryStream.pipe(h264Stream);
   elementaryStream.pipe(this.metadataStream);
 
-  // handle incoming data events
-  h264Stream.on('data', function(data) {
-    // record the track config
-    if (data.nalUnitType === 'seq_parameter_set_rbsp' &&
-        !config) {
-      config = data.config;
-
-      videoTrack.width = config.width;
-      videoTrack.height = config.height;
-      videoTrack.sps = [data.data];
-      videoTrack.profileIdc = config.profileIdc;
-      videoTrack.levelIdc = config.levelIdc;
-      videoTrack.profileCompatibility = config.profileCompatibility;
-    }
-    if (data.nalUnitType === 'pic_parameter_set_rbsp' &&
-        !pps) {
-      pps = data.data;
-      videoTrack.pps = [data.data];
-    }
-  });
-  // generate an init segment based on the first audio sample
-  aacStream.on('data', function(data) {
-    if (audioTrack && audioTrack.channelcount === undefined) {
-      audioTrack.audioobjecttype = data.audioobjecttype;
-      audioTrack.channelcount = data.channelcount;
-      audioTrack.samplerate = data.samplerate;
-      audioTrack.samplingfrequencyindex = data.samplingfrequencyindex;
-      audioTrack.samplesize = data.samplesize;
-    }
-  });
   // hook up the segment streams once track metadata is delivered
   elementaryStream.on('data', function(data) {
-    var triggerData = function(type) {
-      var track = type === 'video' ? videoTrack : audioTrack;
-      return function(mediaData) {
-        // append the media data to an init segment
-        var
-          initSegment = muxjs.mp4.initSegment([track]),
-          segment = new Uint8Array(initSegment.byteLength +
-                                   mediaData.byteLength);
-        segment.set(initSegment);
-        segment.set(mediaData, initSegment.byteLength);
-        self.trigger('data', {
-          type: type,
-          data: segment
-        });
-      };
-    }, i;
+    var i;
+
+    if (data.type === 'audio' || data.type === 'video') {
```

Comment:
Couldn't this be calculated by looking at the audio and video tracks whenever you need it?